src/domain/hoareDomain.ml

```diff
@@ -46,7 +46,8 @@ struct
       else Map.add i b m
   end
 
-  let elements m = Map.values m |> List.of_enum |> List.flatten
+  let elements m =
+    m |> Map.to_seq |> Seq.map snd |> List.of_seq |> List.flatten
```

Comment:
This could be wrong, but it seems like the flattening could be done on `Seq` even:
```suggestion
    m |> Map.to_seq |> Seq.concat_map snd |> List.of_seq
```

Reply:
That has type `'a Seq.t Map.t -> 'a list` instead of `'a list Map.t -> 'a list`.

It would have to be

~~~OCaml
m |> Map.to_seq |> Seq.concat_map (fun x -> List.to_seq @@ snd x) |> List.of_seq
~~~

but round-tripping through Seq for no reason doesn't help here.